packages/repository/src/repository-mixin.ts

```diff
@@ -71,6 +72,26 @@ export function RepositoryMixin<T extends Class<any>>(superClass: T) {
       this.bind(repoKey).toClass(repo);
     }
 
+    /**
+     * Add the datasource to this application.
+     *
+     * @param datasource The datasource to add.
+     *
+     * ```ts
+     *
+     * const ds: juggler.DataSource = new DataSourceConstructor({
+     *   name: 'db',
+     *   connector: 'memory',
+     * });
+     *
+     * app.dataSource(ds);
+     * ```
+     */
+    dataSource(datasource: juggler.DataSource) {
+      const datasourceKey = `datasource`;
```

Comment:
Typically, an application can have multiple datasources. We should build a unique binding key for each datasource, e.g,

```
const datasourceKey = `datasources:${datasource.name}`;
```

Reply:
In package `example-getting-started`, the datasource is created dynamically 
```ts
const datasource =
      this.options && this.options.datasource
        ? new DataSourceConstructor(this.options.datasource)
        : db;
```
When it come to the repository constructor, how should we determine the needed datasource?

Reply:
For `DataSourceConstructor`
```ts
const ds: juggler.DataSource = new DataSourceConstructor({
      name: 'db',
      connector: 'memory',
    });
```
It turns out that `ds.name` equals to `'memory'` not `'db'`

Reply:
> It turns out that ds.name equals to 'memory' not 'db'

This looks like a bug in juggler to me.

I think there should be two ways how to create a named datasource:

```ts
const ds: juggler.DataSource = new DataSourceConstructor('db', {connector: 'memory'});

const ds: juggler.DataSource = new DataSourceConstructor({
  name: 'db',
  connector: 'memory',
});
```

Ideally, both options should return a `DataSource` instance with `ds.name` set to `db`.

Would you like to take a look at this yourself?  I think the changes will be needed in juggler's [lib/datasource.js](https://github.com/strongloop/loopback-datasource-juggler/blob/309b422425a494fea312cf39126d728d7ba4ec46/lib/datasource.js), just remember to add a test verifying your fix (most likely to [test/datasource.js](https://github.com/strongloop/loopback-datasource-juggler/blob/309b422425a494fea312cf39126d728d7ba4ec46/test/datasource.test.js)).

Please mention my GitHub handle in pull request description so that I get a notification to review your changes.